pkg/tsdb/azuremonitor/credentials.go

```diff
@@ -101,7 +104,12 @@ func getAzureCredentials(cfg *setting.Cfg, jsonData *simplejson.Json, secureJson
 	case azcredentials.AzureAuthManagedIdentity:
 		credentials := &azcredentials.AzureManagedIdentityCredentials{}
 		return credentials, nil
-
+	case azcredentials.AzureAuthUserIdentity:
+		credentials := &azcredentials.AzureUserIdentityCredentials{
+			TokenEndpoint: cfg.Azure.UserIdentityTokenEndpoint,
+			AuthHeader:    cfg.Azure.UserIdentityAuthHeader,
```

Comment:
These settings should not be a part of the credential object.

Comment here:
pkg/tsdb/azuremonitor/azcredentials/credentials.go